base/inference.jl

```diff
@@ -3365,40 +3460,44 @@ function optimize(me::InferenceState)
     type_annotate!(me)
 
     # run optimization passes on fulltree
-    force_noinline = false
+    force_noinline = true
```

Comment:
It's a bit of an abuse of the variable name (phrasing it in the positive would be more accurate), since it should be dependent on optimize=true. We use this later to decide whether to compute the inlining optimization for the function. While instead that operation should be separated out into optimization, like the rest of this PR is working towards enforcing.